test/Tests.cmake

```diff
@@ -0,0 +1,38 @@
+# This adds all tests in the test directory.
+include(CTest)
+
+set(TestLib test-lib)
+set(TEST_DIR ${CMAKE_CURRENT_SOURCE_DIR}/test)
+set(TEST_SUFFIX test.c)  # Files that are tests must have names ending with TEST_SUFFIX.
+
+# Add the test files found in DIR to TEST_FILES.
+function(add_test_dir DIR)
+    file(
+        GLOB_RECURSE TEST_FILES_FOR_DIR
+        LIST_DIRECTORIES false
+        RELATIVE ${TEST_DIR}
+        ${DIR}/*${TEST_SUFFIX}
+    )
+    list(APPEND TEST_FILES ${TEST_FILES_FOR_DIR})
+    set(TEST_FILES ${TEST_FILES} PARENT_SCOPE)
+endfunction()
+
+# Add the appropriate directories for the provided build parameters.
+add_test_dir(${TEST_DIR}/general)
+if(NUMBER_OF_WORKERS)
+    add_test_dir(${TEST_DIR}/threaded)
```

Comment:
I personally like `single-threaded` and `multi-threaded` more because `unthreaded` is a bit of a misnomer...

Reply:
Now that you mention it, I see that you are right, at least in principle.

The code often includes the words "unthreaded," "non-threaded," and "threaded." Examples from grep:
```
core/reactor.h: * This file contains header information used by both the threaded and
core/reactor.h: * non-threaded versions of the C runtime. A generated C program will have
core/trace.h: * @param worker The thread number of the worker thread or 0 for unthreaded execution.
core/reactor.c:/* Runtime infrastructure for the non-threaded version of the C target of Lingua Franca. */
core/reactor.c:            tracepoint_reaction_starts(reaction, 0); // 0 indicates unthreaded.
core/reactor_threaded.c: * A struct representing a barrier in threaded 
core/reactor_common.c: * This is the internal implementation shared by both the threaded
core/reactor_common.c: * and non-threaded versions.
```
We could open an issue for this terminology. In principle, it seems like a problem, but there could be reasons for it. For example,
* The "unthreaded" runtime doesn't use a threading library.
* It must be distinguished from the multithreaded runtime in the case where `NUMBER_OF_WORKERS` equals 1.
* "unthreaded" and "threaded" have fewer syllables than "single-threaded" and "multithreaded."

Reply:
Well, this change has been applied -- no more misnomer. Since our use of the word "unthreaded" in other places is a separate issue, it seems okay to mark this one as resolved.